apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Tests/Trace/CloudTraceExtensionTest.cs

```diff
@@ -76,10 +76,14 @@ public void CreateTraceHeaderContext_UseBackUpFunc()
         public void CreateManagedTracer()
         {
             var mockProvider = new Mock<IServiceProvider>();
-            mockProvider.Setup(p => p.GetService(typeof(IHttpContextAccessor))).Returns(new HttpContextAccessor());
+            var mockTracer = new Mock<IManagedTracer>();
+            string traceId = Guid.NewGuid().ToString("N");
+            mockTracer.Setup(p => p.GetCurrentTraceId()).Returns(traceId);
+            ContextTracerManager.SetCurrentTracer(mockTracer.Object);
             var tracer = CloudTraceExtension.CreateManagedTracer(mockProvider.Object);
             Assert.IsType<DelegatingTracer>(tracer);
-            mockProvider.VerifyAll();
+            Assert.Equal(traceId, tracer.GetCurrentTraceId());
+            mockTracer.Verify();
```

Comment:
This `Verify()` does nothing, as no setups were marked with `Verifiable()`. I would drop it completely, as the `Assert.Equal(traceId, ...)` call sufficiently tests the behavior.